zebra-consensus/src/parameters/network_upgrade.rs

```diff
@@ -0,0 +1,163 @@
+//! Network upgrade consensus parameters for Zcash.
+
+use NetworkUpgrade::*;
+
+use std::collections::{BTreeMap, HashMap};
+use std::ops::Bound::*;
+
+use zebra_chain::types::BlockHeight;
+use zebra_chain::{Network, Network::*};
+
+/// A Zcash network protocol upgrade.
+//
+// TODO: are new network upgrades a breaking change, or should we make this
+//       enum non-exhaustive?
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+pub enum NetworkUpgrade {
+    /// The Zcash protocol before the Overwinter upgrade.
+    ///
+    /// We avoid using `Sprout`, because the specification says that Sprout
+    /// is the name of the pre-Sapling protocol, before and after Overwinter.
+    BeforeOverwinter,
+    /// The Zcash protocol after the Overwinter upgrade.
+    Overwinter,
+    /// The Zcash protocol after the Sapling upgrade.
+    Sapling,
+    /// The Zcash protocol after the Blossom upgrade.
+    Blossom,
+    /// The Zcash protocol after the Heartwood upgrade.
+    Heartwood,
+    /// The Zcash protocol after the Canopy upgrade.
+    Canopy,
+}
+
+/// Mainnet network upgrade activation heights.
+///
+/// This is actually a bijective map, but it is const, so we use a vector, and
```

Comment:
TIOLI: being super pedantic
```suggestion
/// This is actually a bijective map, but it is const, so we use a slice, and
```